modules/shortcodes/tweet.php

```diff
@@ -15,11 +15,24 @@
  * More parameters and another tweet syntax admitted:
  * [tweet tweet="https://twitter.com/jack/statuses/20" align="left" width="350" align="center" lang="es"]
  *
+ * Tweets can also be referenced without a shortcode:
+ * https://twitter.com/jack/statuses/20
+ *
  * @package Jetpack
  */
 
 add_shortcode( 'tweet', array( 'Jetpack_Tweet', 'jetpack_tweet_shortcode' ) );
 
+/**
+ * Prevent single line Tweets from being incorrectly parsed as markdown
+ */
+function twitter_custom_md_pattern( $patterns ) {
+    $patterns[] = '/^http(s|):\/\/twitter\.com(\/\#\!\/|\/)([a-zA-Z0-9_]{1,20})\/status(es)*\/(\d+)$/';
```

Comment:
It seems Twitter usernames can be 15 characters maximum:
https://help.twitter.com/en/managing-your-account/twitter-username-rules#:~:text=Usernames%20containing%20the%20words%20Twitter,for%20the%20sake%20of%20ease.

Reply:
I just copied your existing regex, which has 20.
 https://github.com/Automattic/jetpack/blob/503dbc4cd75d6859e2ab1d524fc7e337122f4b9c/modules/shortcodes/tweet.php#L73
I can update it if you like?